### PR TITLE
Fix doubleclick on tree item, restore input focus on previous windows.

### LINKED ISSUE #37335: SceneTreeDock loses focus when changing Node's name & db-click not working
**Godot version:**
Current master branch (with DisplayServer update to separate windows)

**OS/device including version:**
Fedora
**Issue description:**
When double-click on Node to rename it - it doesn't popup TextBox with Node's name.
Only Enter key works - but then, after name is typed, SceneTreeDock is unfocused, 
so it's not possible to use arrow keys to go into next Node
**Steps to reproduce:**
Try to double click on Node.
Try to move Up or Down after you Enter-keyed renamed the Node

scenetreedock

### PR DESCRIPTION
**Pull request by reduz**
Closes #37335

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]

File: scene/main/viewport.cpp [hunk 1]
```diff
@@ -3303,6 +3321,7 @@ void Viewport::_bind_methods() {
 
 	ClassDB::bind_method(D_METHOD("_gui_show_tooltip"), &Viewport::_gui_show_tooltip);
 	ClassDB::bind_method(D_METHOD("_gui_remove_focus"), &Viewport::_gui_remove_focus);
+	ClassDB::bind_method(D_METHOD("_gui_remove_focus_for_window"), &Viewport::_gui_remove_focus_for_window);
```
Comment (akien-mga) [inline scene/main/viewport.cpp:3324]: It seems both `_gui_show_tooltip` and `_gui_remove_focus` don't need to be bound, nothing seems to depend on them being bound (unless they're meant to be used from scripting, but then they shouldn't start with `_`).

Comment (bruvzg) [reply scene/main/viewport.cpp:3322]: `_gui_remove_focus` is called by name here:

https://github.com/godotengine/godot/blob/5f2295f2dfdbf1ac4e63b562e4cef71f2fda4036/scene/main/viewport.cpp#L2470

Comment (akien-mga) [reply scene/main/viewport.cpp:3322]: Fixed.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Needs rebase after #40015.

File: scene/main/viewport.cpp (hunk 1 above)
Comment (akien-mga) [reply scene/main/viewport.cpp:3322]: @bruvzg Not after this commit :)

**Comment by akien-mga** [comment]
Thanks!

File: scene/main/viewport.cpp [hunk 2]
```diff
@@ -2450,6 +2450,22 @@ void Viewport::_gui_remove_control(Control *p_control) {
...
+	Viewport *v = const_cast<Viewport *>(this);
+	Window *w = Object::cast_to<Window>(v);
+	while (!w) {
+		v = v->get_parent_viewport();
+		w = Object::cast_to<Window>(v);
+	}
+
+	return w;
+}
+void Viewport::_gui_remove_focus_for_window(Node *p_window) {
```
Comment (Anutrix) [inline scene/main/viewport.cpp:2463]: Maybe add an empty line between the functions.

**Pull request merged**